desc/objectives/_geometry.py

```diff
@@ -711,8 +711,8 @@ def build(self, use_jit=True, verbose=1):
                 params=self._surface.params_dict,
                 transforms=surface_transforms,
                 profiles={},
-                basis="xyz",
             )["x"]
+            surface_coords = rpz2xyz(surface_coords)
```

Comment:
why is this needed? can't you still request `basis=xyz`?

Reply:
Inside the objectives, we don't use `compute` but instead use `_compute` which is jittable. Since the latter is recursive, I put conversion to the former. So, only for objective cases, you should change basis manually

Reply:
Got it, thanks. Might be worth noting this in the [making an objective function tutorial in the docs](https://desc-docs.readthedocs.io/en/latest/adding_objectives.html) then

Reply:
It is documented in the `compute` and `_compute` functions